less/core/nav.less

```diff
@@ -24,16 +24,60 @@ html.is-nav-bottom #wrapper {
   &__inner {
     .l-container-responsive(@max-page-width);
     display: flex;
-    align-items: center;
+    align-items: stretch;
   }
 
-  // Override floats on additional `nav__btn` elements part of plugins
-  // that haven't been updated to work with data-order yet.
-  // Can be removed in time.
   &__btn {
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+
+    // Override floats on additional `nav__btn` elements part of plugins
+    // that haven't been updated to work with data-order yet.
+    // Can be removed in time.
     float: none !important;
   }
 
+  &__btn-label {
+    display: none;
+  }
+
+  &.show-label {
+    .show-nav-label(small);
+    .show-nav-label(medium);
+    .show-nav-label(large);
+  }
+
+  &.has-label-right &__btn-label {
+    margin: 0 0 0 0.5rem;
+  }
+
+  &.has-label-bottom {
+    .nav__btn {
```

Comment:
Keen to get your thoughts on two entries rather than nesting:

```
&.has-label-bottom &__btn {}

&.has-label-bottom &__btn-label {}
```

Reply:
I'm fine with the nesting, but I don't have strong feelings either way.

Nesting allows for code folding in an editor, which is nice, and it also groups the styles for each label position in the same code block.

<img src="https://user-images.githubusercontent.com/898168/233678424-6031ca50-c9d0-4ca0-bc1d-d8c88081858b.jpg" width="350">

Reply:
I can certainly see the benefit of this approach.